### PR TITLE
Add locale types

### PR DESCRIPTION
**Pull request by forivall**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/locale/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// Type definitions for locale 0.1
+// Project: https://github.com/florrain/locale
+// Definitions by: Emily Marigold Klassen <https://github.com/forivall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = locale;
+
+declare global {
+    namespace Express {
+        interface Request {
```
Comment (amcasey) [inline types/locale/index.d.ts]: Sorry to be obtuse, but I'm not sure what this is for.  Could you maybe add a sample usage in the test file?

**Review by amcasey — CHANGES_REQUESTED** [review]
I'm missing something with the global augmentation.  Otherwise, LGTM.

File: types/locale/index.d.ts (hunk 1 above)
Comment (forivall) [reply types/locale/index.d.ts]: The test file has a usage at https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8ec0ab5eb0283bb7ef715ec3b4b68f11e1f93cd0/types/locale/locale-tests.ts#L44 - See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L19 for info. Most declarations that use this functionality don't have anything to describe what this global augmentation does, but `passport` does have the line `// These declarations are merged into express's Request type`

Comment (amcasey) [reply types/locale/index.d.ts]: Thanks for the explanation!

**Review by amcasey — APPROVED** [review]

**Pull request merged**